### DISCUSSION TITLE
Possible to Install Turbo without npm or node?

### DISCUSSION THREAD

**Discussion by jbcpollak** [vercel/turborepo#9278] [Help] [answered]
### Summary

This is definitely "off book" so I'd understand if the answer is a blanket no, but I just wanted to check - we use Turbo to build an entirely non-JS project, because we like the simplicity of setup and the caching vs something like Bazel.

For our use case we run inside a dev container and the only reason we install nodejs in the container is to install Turbo.

Is there any way we could just jump to the rust crate a install that, or is there wrapper JS that is needed to use run Turbo?

Thanks.

**Comment by anthonyshew** [comment, marked as answer]
We've included notes on this in our [Support Policy](https://turbo.build/repo/docs/getting-started/support-policy#nodejs).

The bit you likely care most about:
> Core turbo functionality does not depend on the active Node.js version on your system, but [some packages in the Turborepo ecosystem](https://turbo.build/repo/docs/reference#packages) such as [create-turbo](https://turbo.build/repo/docs/reference/create-turbo), [turbo-ignore](https://turbo.build/repo/docs/reference/turbo-ignore), and [eslint-config-turbo](https://turbo.build/repo/docs/reference/eslint-config-turbo) do.

**Reply by jbcpollak** [reply]
Thanks for the reference. The policy also says:

> Core turbo functionality depends on the package managers in the JavaScript ecosystem and their implementations of Workspaces and lockfiles formats

Does this mean `turbo` requires a package manager to be installed?

Is there a source for downloading the released core turbo binaries independent of an npm package? (Ie, install it without `npm i`)?

Thanks

**Reply by anthonyshew** [reply]
We look in the lockfile for specific pieces of information about the relationships between your packages, though this isn't necessarily a requirement.

Yes, Turborepo is distributed through the npm registry, but you could also download source and build it yourself.
